msal/src/test/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt

```diff
@@ -181,8 +214,19 @@ class NativeAuthPublicClientApplicationKotlinTest : PublicClientApplicationAbstr
             MockApiResponseType.TOKEN_SUCCESS
         )
 
+        // check the safe logging
+        val latch = CountDownLatch(1)
+        val loggerCallback = LoggerTestCallback(allowPIITrueToCheck, allowPIIFalseToCheck, allowPII, latch)
+        Logger.getInstance().setExternalLogger(loggerCallback)
+
         val result = application.signIn(username, password)
+
+        withContext(Dispatchers.IO) {
+            latch.await()
```

Comment:
Why is this happening on a coroutine? Unless there's good reason to, I wouldn't be mixing coroutine with callbacks. But moreover, I don't see the reason for this `await()` to happen on a different thread. The whole point is that the main thread waits.

Reply:
Remove the  withContext(Dispatchers.IO) here instead of 
```kotlin
val shouldLogBeDiscarded = false
        val timedOut: Boolean = !latch.await(DISCARDED_TIME_IN_MILLISECONDS, TimeUnit.MILLISECONDS)
        assertEquals(shouldLogBeDiscarded, timedOut)
```
Similar like LoggerTest.java